### PR TITLE
Fix/issue 1746

### LINKED ISSUE #1746: Improve default value reporting in `--help`
### Description
As a user, I want `--help` to show default values for all options where it makes sense, especially the boolean ones, so that I know exactly what is enabled and what isn't.

For example `--eth1-enabled` defaults to `true`, whereas `--rest-api-docs-enabled` seems to default to `false`. But there is no way to know this.

(Related: although the docs for `--rest-api-docs-enabled` in [the code](https://github.com/PegaSysEng/teku/blob/81deceb0070d240768fe3a43fa833fbb3604e9e2/teku/src/main/java/tech/pegasys/teku/cli/options/BeaconRestApiOptions.java#L27) seem to give `true` for the `fallbackValue`, the default appears to be `false`.)

### Acceptance Criteria
* running `teku --help` shows the correct default settings for all relevant options.

**Expected behavior:** [What you expect to happen]
```
      --eth1-enabled[=<BOOLEAN>]
                             Whether to connect to the ETH1 chain
                             Default: true
```

**Actual behavior:** [What actually happens]
```
      --eth1-enabled[=<BOOLEAN>]
                             Whether to connect to the ETH1 chain
```

### Versions (Add all that apply)
* Software version: [`teku --version`]
  - teku/v0.11.0-dev-e3ad800d/linux-x86_64/-privatebuild-openjdk64bitservervm-java-13

### PR DESCRIPTION
**Pull request by ermyas**
## PR Description
The usage help message of a CLI command should show the default values of its options if applicable. However, currently, all commands/subcommands do not display the defaults of their boolean options. This PR addresses this issue.

## Fixed Issue(s)

Fixes #1746 

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

The [CLI reference documentation](https://docs.teku.consensys.net/en/latest/Reference/CLI) is a bit out of sync with the code. Below are some of the minor discrepancies that need to be addressed in the documentation

*[Teku command - `teku`](https://docs.teku.consensys.net/en/latest/Reference/CLI/CLI-Syntax/)*
— `--log-file-name-pattern` documentation should perhaps show the default pattern used `<data-path>/logs/teku_%d{yyyy-MM-dd}.log`
— `—metrics-categories` missing `DISCOVERY` category (i.e. `TekuMetricCategory.DISCOVERY`)
— `—validators-external-signer-timeout` defaults to 5000 in code, but docs state 1000
—  `--p2p-private-key-file`  documentation should describe what happens when this value is not provided.

*[Slashing-protection subcommand - `teku slashing-protection`](https://docs.teku.consensys.net/en/latest/Reference/CLI/Subcommands/Slashing-Protection/)*
— Documentation for `ImportCommand` and `ExportCommand` should  mention `—data-base-path` as an alias to `—data-path` 
— `data-validator-path` option not present in the documentation

*[Validator-client subcommand - `teku validator-client`](https://docs.teku.consensys.net/en/latest/Reference/CLI/Subcommands/Validator-Client/)*
— `—log-include-validator-duties-enabled` default to `true` in code but the docs report the default as `false`  [Validator client subcommand options](https://docs.teku.consensys.net/en/latest/Reference/CLI/Subcommands/Validator-Client/#log-include-events-enabled) 
—`--log-file-name-pattern` documentation should show the default pattern used `<data-path>/logs/teku_%d{yyyy-MM-dd}.log`
— `—metrics-categories` missing `DISCOVERY` category
— `—metrics-interface` defaults to “127.0.0.1” in code, but docs note “0.0.0.0” 
— `--validators-early-attestations-enabled` option not present in the documentation
— `--validators-external-signer-keystore` and `--validators-external-signer-keystore-password-file` options not present in the documentation
— `—validators-external-signer-timeout` defaults to 5000 in code, but docs state 1000
—`--validators-external-signer-truststore` and `—--validators-external-signer-truststore-password-file` options not present in the documentation
— `--validators-graffiti-file` option not present in the documentation

*[Voluntary-exit subcommand -- `teku voluntary-exit`](https://docs.teku.consensys.net/en/latest/Reference/CLI/Subcommands/Voluntary-Exit/)*
— `--validators-external-signer-keystore` and `--validators-external-signer-keystore-password-file` options not present in the documentation
—`--validators-external-signer-truststore` and `—--validators-external-signer-truststore-password-file` options not present in the documentation
— `—validators-external-signer-timeout` defaults to 5000 in code, but docs state 1000

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: teku/src/main/java/tech/pegasys/teku/cli/options/DataStorageOptions.java [hunk 1]
```diff
@@ -46,6 +48,7 @@
   @CommandLine.Option(
       names = {"--Xdata-storage-non-canonical-blocks-enabled"},
       paramLabel = "<BOOLEAN>",
+      showDefaultValue = CommandLine.Help.Visibility.ALWAYS,
```
Comment (ajsutton) [inline teku/src/main/java/tech/pegasys/teku/cli/options/DataStorageOptions.java]: nit: Can we add the import so this is just `Visibility.ALWAYS` for consistency?

File: teku/src/main/java/tech/pegasys/teku/cli/options/MetricsOptions.java [hunk 2]
```diff
@@ -35,6 +36,7 @@
   @Option(
       names = {"--metrics-enabled"},
       paramLabel = "<BOOLEAN>",
+      showDefaultValue = CommandLine.Help.Visibility.ALWAYS,
```
Comment (ajsutton) [inline teku/src/main/java/tech/pegasys/teku/cli/options/MetricsOptions.java]: nit: Can we add the import so this is just `Visibility.ALWAYS` for consistency?

File: teku/src/main/java/tech/pegasys/teku/cli/options/P2POptions.java [hunk 3]
```diff
@@ -66,21 +69,24 @@
...
   private Integer p2pAdvertisedPort;
 
   @Option(
       names = {"--p2p-private-key-file"},
       paramLabel = "<FILENAME>",
-      description = "This node's private key file",
+      description =
+          "This node's private key file. If one is not specified, the system looks for a previously generated key in the key-value store "
+              + "(i.e. <beacon-data-dir>/kvstore/generated-node-key.dat) and uses that if available. "
+              + "If not, then a new key is generated for use and persisted in the key-value store.",
```
Comment (ajsutton) [inline teku/src/main/java/tech/pegasys/teku/cli/options/P2POptions.java]: We probably don't want to expose the inner details of how the beacon node store's its data.  So maybe:
```suggestion
          "This node's private key file. If not specified, uses or generates a key which is stored within the <beacon-data-dir>.",
```

Comment (ajsutton) [reply teku/src/main/java/tech/pegasys/teku/cli/options/P2POptions.java]: @bgravenorst Any chance you have a nice succinct way of phrasing this?

File: teku/src/main/java/tech/pegasys/teku/cli/options/ValidatorOptions.java [hunk 4]
```diff
@@ -41,7 +42,9 @@
       names = {"--validators-graffiti-file"},
       paramLabel = "<GRAFFITI FILE>",
       description =
-          "File to load graffiti value to include during block creation. Value gets converted to bytes and padded to Bytes32.  If file reading fails during block creation, teku will fall back to any value supplied via --validators-graffiti.",
+          "File to load graffiti value to include during block creation. Value gets converted to bytes and padded to Bytes32."
+              + "This option takes precedence over --validators-graffiti, but if file reading fails during block creation, "
+              + "teku will fall back to any value supplied via --validators-graffiti.",
```
Comment (ajsutton) [inline teku/src/main/java/tech/pegasys/teku/cli/options/ValidatorOptions.java]: Maybe simplify to:
```suggestion
          "File to load graffiti value to include during block creation. Value gets converted to bytes and padded to Bytes32. "
              + "Takes precedence over --validators-graffiti. If the file can not be read, the --validators-graffiti value is used as a fallback.",
```

File: teku/src/main/java/tech/pegasys/teku/cli/subcommand/VoluntaryExitCommand.java [hunk 5]
```diff
@@ -95,6 +96,7 @@
       names = {"--confirmation-enabled"},
       description = "Request confirmation before submitting voluntary exits.",
       paramLabel = "<BOOLEAN>",
+      showDefaultValue = CommandLine.Help.Visibility.ALWAYS,
```
Comment (ajsutton) [inline teku/src/main/java/tech/pegasys/teku/cli/subcommand/VoluntaryExitCommand.java]: nit: Use the import here too.

File: teku/src/test/java/tech/pegasys/teku/cli/BeaconNodeCommandTest.java [hunk 6]
```diff
@@ -96,6 +98,36 @@ public void helpOptionShouldDisplayFullHelp() {
     assertThat(str).doesNotContain("To display full help:");
   }
 
+  @Test
+  public void helpShouldShowDefaultsForBooleanOptions() {
```
Comment (ajsutton) [inline teku/src/test/java/tech/pegasys/teku/cli/BeaconNodeCommandTest.java:102]: This test is genius.  Nice work.

**Review by ajsutton — COMMENTED** [review]
This is looking really good.  I've just left a few small suggestions but it's a huge improvement.

Thanks for taking this on.

File: teku/src/main/java/tech/pegasys/teku/cli/options/P2POptions.java (hunk 3 above)
Comment (bgravenorst) [reply teku/src/main/java/tech/pegasys/teku/cli/options/P2POptions.java]: This is a bit wordier, but how about, "The node's private key file. Defaults to the key file in `<beacon-data-dir>`. If no key file exists, Teku creates a key file containing the generated private key."?

File: teku/src/main/java/tech/pegasys/teku/cli/options/DataStorageOptions.java (hunk 1 above)
Comment (ermyas) [reply teku/src/main/java/tech/pegasys/teku/cli/options/DataStorageOptions.java]: sure, no worries. 
I have been wondering about this particular option though (i.e. `--Xdata-storage-non-canonical-blocks-enabled`). Can I confirm that it is indeed meant to be visible? I guess all other options prefixed with '--X' (unstable options?) are hidden except for this.

File: teku/src/main/java/tech/pegasys/teku/cli/options/P2POptions.java (hunk 3 above)
Comment (ermyas) [reply teku/src/main/java/tech/pegasys/teku/cli/options/P2POptions.java]: > We probably don't want to expose the inner details …

makes sense. 

@ajsutton's suggestion feels a bit more concise, though @bgravenorst's communicates the sequence of default behaviors a bit more clearly. I might apply @ajsutton's suggestion for now, and we can change it if you feel different. I guess we can probably be more descriptive in the CLI reference docs instead if we feel there is still room for ambiguity.

File: teku/src/main/java/tech/pegasys/teku/cli/options/DataStorageOptions.java (hunk 1 above)
Comment (ajsutton) [reply teku/src/main/java/tech/pegasys/teku/cli/options/DataStorageOptions.java]: hmm, I'm not sure but it's definitely wrong.  It should either be hidden or not have a `--X` prefix but I'm not exactly sure which it should be.  I think it shouldn't have the --X prefix.  @rolfyone any idea?

**Review by ajsutton — APPROVED** [review]
LGTM. Thank you!

File: teku/src/main/java/tech/pegasys/teku/cli/options/DataStorageOptions.java (hunk 1 above)
Comment (ajsutton) [reply teku/src/main/java/tech/pegasys/teku/cli/options/DataStorageOptions.java]: Will merge this and leave any potential clean up here for a separate PR.

**Pull request merged**